XiEditor/EditViewController.swift

```diff
@@ -505,19 +539,41 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
         }
         dragEvent = theEvent
     }
-    
+
     override func mouseUp(with theEvent: NSEvent) {
         dragTimer?.invalidate()
         dragTimer = nil
         dragEvent = nil
     }
-    
+
+    override func mouseMoved(with theEvent: NSEvent) {
+        if !editView.isFirstResponder {
+            editView.window?.makeFirstResponder(editView)
+        }
+        if hoverTimer == nil && theEvent.modifierFlags.contains([.option]){
+            hoverTimer = Timer.scheduledTimer(timeInterval: TimeInterval(3.0), target: self, selector: #selector(sendHover), userInfo: nil, repeats: false)
+        }
+        hoverEvent = theEvent
+    }
+
+    @objc func sendHover() {
+        if !infoPopover.isShown {
+            if let event = hoverEvent {
+                let hoverPosition = editView.bufferPositionFromPoint(event.locationInWindow)
+                hoverTimer?.invalidate()
+                hoverTimer = nil
+                document.sendRpcAsync("request_hover", params: ["request_id": hoverRequestID, "position": ["type": "utf8_line_char", "line": hoverPosition.line, "character": hoverPosition.column]])
```

Comment:
If we're going to use this in two places, I would add a method to `Document` that handles the implementation.

I believe that the RPC expected in core has changed as well; we don't send `type` and we send `column` instead of `character`.